CMakeLists.txt

```diff
@@ -14,8 +14,17 @@
 #                                                                         #
 ###########################################################################
 
+
+set(CUDA_ENABLED false)
+
 # Mandatory call to project
-project(opm-simulators C CXX)
+if(CUDA_ENABLED)
+  project(opm-simulators LANGUAGES CUDA C CXX)
+  # cuda_runtime.h
+  include_directories(/path/to/cuda/include)
```

Comment:
Whether we need a  CUDA_ENABLED switch is maybe a matter of taste. Up to now I was not aware that one could specify CUDA as a language to `project`. If that imlicitly run find_package(CUDA) one coulde use the `-DCMAKE_DISABLE_FIND_PACKAGE_CUDA=ON` switch  to disable cuda.

It seems like CUDA was added as a language from version 3.12 but we are supporting Ubuntu bionic which only ship the 3.10 version. Maybe we should leave the `project` as is and run `find_package(CUDA)` afterwards or even better specify it in the opm-simulators-prereqs.cmake file.

Also the include path used seems to be wrong on most systems.  Maybe we should use the `CUDA_INCLUDE_DIRECTORIES` variable instead?

Reply:
I think using `find_package(CUDA)` is much better. I'll try adding a HAVE_CUDA to the opm-simulators-prereqs.cmake file. This will also solve your second point, because .cpp files are indeed not compiled by `nvcc`.